### ISSUE TITLE
Using Custom Score and Loss Functions

### ISSUE THREAD

**Issue by darraghdog** [dnouri/nolearn#186]
Hi, I am trying to use the custom_score or objective_loss_function on a classification problem. I stripped the problem down to show it in the toy examples provided in nolearn. Below can be seen, without the custom_score or objective_loss_function, 2 class classifciation works fine. However when I add a custom score or loss function it fails. Can you advise where I am going wrong. 

I am using : 
 lasagne  0.2.dev1
theano  0.7.0.dev-a85a44fc70d9bf8faff0d7f91a03127af582d015

![custom](https://cloud.githubusercontent.com/assets/7209588/12057962/a42cb874-af3e-11e5-82a9-a0b115de0ea8.png)

**Comment by BenjaminBossan** [comment]
Hi, those are two separate problems.

1) sklearn's `roc_auc_score` expects just one column but gets 2, try something like this:

    custom_score=('auc', lambda y_true, y_proba: roc_auc_score(y_true, y_proba[:, 1])

Regarding 2), I will need to have a closer look but the error already indicates that it's also a shape problem. Does binary cross-entropy make sense with sklearn's `make_classification`, i.e. is it a binary classification?

**Comment by darraghdog** [comment]
Thanks Benjamin, for the custom scoring, it worked out... see below.

Regarding the custom objective, the problem I am working on is binary classifiction. I had the mis-shape issue on the problem I am currently working, so I wanted to try it out on the toy data set provided with Lasagne/nolearn to try and discount the chance of the problem coming from any of my code or data.

The measurement I have is auc, so I wanted to test out nnets using some of the binary custom objectives provided. I understand if I do not define a custom objective, Lasagne will default to squared error.  

![screenshot from 2016-01-01 21 16 47](https://cloud.githubusercontent.com/assets/7209588/12072241/0220d552-b0cd-11e5-948e-fccdb7020463.png)

**Comment by BenjaminBossan** [comment]
nolearn defaults to `categorical_crossentropy` when not explicitly setting `regression=True`. Cross-entropy should be fine when trying to minimize AUC. From my experience, when trying to minimize AUC, playing with the loss function is not worth it, better try feature engineering etc.

**Comment by darraghdog** [comment]
I will work with categorical_crossentropy and close the issue. Thanks for the feedback.
